074-connector-offsets-support.md

```diff
@@ -0,0 +1,180 @@
+* Monitoring the progress of running connectors on a per-source-partition basis 
+* Skipping records that cause issues with a connector and cannot be addressed using existing error-handling features
+
+## Proposal
+
+Strimzi should include mechanisms to let users manage offsets when utilising the KafkaConnector resources.
+This proposal suggests different approaches for each action, since they are all slightly different.
+In summary:
+ - listing offsets via the KafkaConnector status and enabled/disabled through a property
+ - altering offsets via a new resource called KafkaConnectorOffsets
+ - deleting offsets via an annotation on the KafkaConnector resource
+
+### Request/Response from the API endpoints
+
+The request/response used when interacting with the Kafka Connect API is different for source and sink connectors.
+This is because sink connectors use regular Kafka offsets, whereas source connectors determine their own structure for offsets.
+This difference is reflected in the approaches outlined in this proposal.
+
+Source connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        // Connector-defined source partition
+      },
+      "offset": {
+        // Connector-defined source offset
+      }
+    }
+  ]
+}
+```
+
+Sink connector format is:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "kafka_topic": // Kafka topic
+        "kafka_partition": // Kafka partition
+      },
+      "offset": {
+        "kafka_offset": // Kafka offset
+      }
+    }
+  ]
+}
+```
+
+### Listing offsets
+The KafkaConnector resource will be updated with a new property to enable the current offsets to be listed in the resource status.
+
+The new property will be a boolean property called `showOffsets` and default to `false`. It should default to `false` because enabling it requires the operator to do an additional API call every time it reconciles the connector.
+If there are a lot of connectors running in the cluster this is a lot of additional API calls.
+
+When the property is set to `true` the status will include the offsets.
+
+The details in the status will be similar to the format returned by the Connect API:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnector
+#...
+status:
+  #...
+  connectorOffsets:
+    lastUpdateTime: "2024-05-21T09:08:58.467712198Z"
+    source:
+      - partition: "{\"connector-determined-partition\": \"1\"}" # value is a String containing JSON as received from the API
+        offset: "{\"connector-determined-offset\": \"1\"}" # value is a String containing JSON as received from the API
+    sink:
+      - partition:
+          kafkaTopic: "my-topic"
+          kafkaPartition: 1
+        offset:
+          kafkaOffset: 0
+```
+
+Notes:
+* Only one of the `source` and `sink` properties will be present.
+* The operator will update the `lastUpdateTime` whenever it updates the offsets section of the status.
```

Comment:
When you say "to not update it just because of the timestamp" do you mean to not trigger a never-ending loop liked discussed earlier in the proposal?

Reply:
Yeah. With updating the timestamp even without the offset change just to indicate that at this time these offsets still applied it will be just the timestamp change triggering the loop. So you will have a busy loop even for connectors that are almost idle. That is what I meant I think.